packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -379,7 +379,8 @@ implements ISerializableInterval {
         const newInterval =
             createSequenceInterval(label, startPos, endPos, this.start.getClient(), this.intervalType, op);
         if (this.properties) {
-            newInterval.addProperties(this.properties);
+            newInterval.properties = createMap<any>();
+            this.propertyManager.copyTo(this.properties, newInterval.properties, newInterval.propertyManager);
```

Comment:
the change incurred by `modify` has to do with the interval's endpoints, not its properties though. Regardless of whether it was requested locally or remotely, we want the "modified" interval to have the same PropertyManager state.

Reply:
Doesn't look like it marks all props as pending to me:

```typescript
    public copyTo(
        oldProps: PropertySet,
        newProps: PropertySet | undefined,
        newManager: PropertiesManager,
    ): PropertySet | undefined {
        if (oldProps) {
            if (!newProps) {
                // eslint-disable-next-line no-param-reassign
                newProps = createMap<any>();
            }
            if (!newManager) {
                throw new Error("Must provide new PropertyManager");
            }
            for (const key of Object.keys(oldProps)) {
                newProps[key] = oldProps[key];
            }
            newManager.pendingRewriteCount = this.pendingRewriteCount;
            newManager.pendingKeyUpdateCount = createMap<number>();
            for (const key of Object.keys(this.pendingKeyUpdateCount!)) {
                newManager.pendingKeyUpdateCount[key] = this.pendingKeyUpdateCount![key];
            }
        }
        return newProps;
    }
```

Looks like it just sets up assigning to all fields of `newProps` from `oldProps`' values, then copies over the pending local state to the new property manager. Pending local state representation is per-key and counter-based, so that remote op processing only incurs changes if there aren't local changes on that key (and additionally there aren't any local "rewrite"s, though I'm not sure that's exposed for intervals).

Simple example not involving rewrite semantics: client 1 adds `{ foo: "foo1", bar: "bar2"}`, then adds `{ foo: "foo3" }`. I'd expect its property manager to have the following state:

```{
  properties: { foo: "foo3", bar: "bar2" },
  pendingKeyUpdateCount: { foo: 2, bar: 1 }
}```